compiler/src/dotty/tools/dotc/cc/CheckCaptures.scala

```diff
@@ -507,14 +507,29 @@ class CheckCaptures extends Recheck, SymTransformer:
       recheckFinish(result, arg, pt)
     */
 
-    /** If expected type `pt` is boxed, don't propagate free variables.
+    extension (tree: Tree)
+      private def isFunctionLiteral(using Context): Boolean = tree match
+        case Block((defTree @ DefDef(_, _, _, _)) :: Nil, Closure(_, meth, _)) =>
+          val defSym = defTree.symbol
+          val methSym = meth.symbol
+          defSym.eq(methSym)
+        case _ =>
+          false
+
+      private def isRefTree: Boolean = tree match
```

Comment:
Just use `isInstanceOf[RefTree]` instead, that avoids having to define a new function.